src/Bits.hs

```diff
@@ -31,3 +31,8 @@ nor x y = not (or x y)
 
 xnor :: Bit -> Bit -> Bit
 xnor x y = not (xor x y)
+
+implies :: Bit -> Bit -> Bit
+implies I I = I
+implies I O = O
+implies O _ = I
```

Comment:
Since you already said in the test what it does, why not implement it exactly that way?
```suggestion
implies :: Bit -> Bit -> Bit
implies I O = O
implies _ _ = I
```

That of course makes the test trivial to the point of being nonsensical, but that only says something about how powerful Haskell is.

The test is still valuable though if we wanted to change implementation later, say to express everything in terms of `NAND`, like in a real computer.